### PR TITLE
Print memory stats after start_step in train

### PR DESCRIPTION
**Pull request by hengtaoguo**
# Description

Display the memory statistics (including percentages) for all utilized devices. In `max_utils.py`, replace `print` statements with `max_logging` to maintain consistency.

FIXES: b/385169408

# Tests

Ran tests on local v4-8 devbox and xpk workload on v5e-512 cluster.
image
memstats

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: MaxText/train.py [hunk 1]
```diff
@@ -935,6 +935,9 @@ def train_loop(config, state=None):
         jax.block_until_ready(state)  # Block until current state finishes to end profile cleanly
       prof.deactivate()
 
+    if step == start_step:
+      max_utils.print_mem_stats(label=config.run_name)
```
Comment (gobbleturk) [inline MaxText/train.py:939]: Can we change label here to "After params initalized" as well like https://github.com/AI-Hypercomputer/maxtext/blob/6ec3368af31fff6e6d735ac9d5fb77f91fc0f784/MaxText/maxengine.py#L156?

For reference this is only showing the HBM usage of the optimizer state. This can help us verify that host offloading is working as expected

Comment (hengtaoguo) [reply MaxText/train.py]: Thanks for the suggestion and done!

**Review by gobbleturk — APPROVED** [review]
Thanks hengtao!
